linkis-computation-governance/linkis-engineconn/linkis-computation-engineconn/src/main/scala/org/apache/linkis/engineconn/computation/executor/hook/UDFLoadEngineConnHook.scala

```diff
@@ -174,11 +174,11 @@ abstract class UDFLoad extends Logging {
     logger.info(s"Successful to execute function code ${runType}, type : ${udfType}")
   }
 
-  protected def loadUDF(labels: Array[Label[_]]): Unit = {
+  protected def loadUDF(): Unit = {
 
     val codes = getFunctionCode()
     if (null != codes && codes.nonEmpty) {
-      val executor = ExecutorManager.getInstance.getExecutorByLabels(labels)
+      val executor = ExecutorManager.getInstance.getReportExecutor
```

Comment:
The Executor here needs to be obtained through Labels, and the default Executor cannot be used, because UDF supports different languages. For Spark, there are different UDF registration languages such as SQL, Python, and Scala.

Reply:
Many thx for your remind! 
I infer from the code that AbstractEngineConnExecution has already created one executor by labels as depicted by codes below:
```
override def getReportExecutor: OnceExecutor = if (getExecutors.isEmpty) {
    val labels =
      if (null == engineConn.getEngineCreationContext.getLabels()) {
        throw new OnceEngineConnErrorException(12560, "No labels is exists.")
      } else engineConn.getEngineCreationContext.getLabels()
    createExecutor(
      engineConn.getEngineCreationContext,
      labels.toArray[Label[_]](Array.empty[Label[_]])
    ).asInstanceOf[OnceExecutor]
  } else {
    getExecutors
      .filter(_.isInstanceOf[OnceExecutor])
      .maxBy {
        case executor: SensibleExecutor => executor.getStatus.ordinal()
        case executor: Executor => executor.getId.hashCode
      }
      .asInstanceOf[OnceExecutor]
  }
```
After creation, the following execution hooks(such as UDFLoadEngineConnHook) should do register UDF stuff. Just as the issue illustrated, the hook created another executor which throws exception to hinder EngineConnServer to start successfully. My PR wants the hook to reuse the executor created before(The previously created executor has already been created by labels). For Spark, different UDF registration languages such as SQL, Python, and Scala must be with different labels. So I should correct my codes again=> UDFLoadEngineConnHook should get executor with full labels instead of just by codeLanguageLabel(which would cause the issue). Am I right? @peacewong

Reply:
@peacewong I have recorrected my PR according to what I have explained above.

Reply:
@QuantumXiecao Thanks for your update, now the Spark EngineConnUDF registration will register all types of UDFs when EC starts. Because Spark supports multiple Executors, for example, Python Executors can use UDFs registered with Sql and Scala

Reply:
@peacewong 
After revisiting codes constantly, I have finally got the point: flink engine conn now do not support udf.
So configing the JarUdfEngineHook would cause the issue.
By the way, I think in the future we should also support udf for flink in linkis.

Reply:
@QuantumXiecao Yes, this has been planned into the RoadMap in 2023, now you can remove the corresponding hook